pydantic_ai_slim/pydantic_ai/models/google.py

```diff
@@ -770,6 +787,65 @@ async def _get_event_iterator(self) -> AsyncIterator[ModelResponseStreamEvent]:
                 else:
                     assert part.function_response is not None, f'Unexpected part: {part}'  # pragma: no cover
 
+    async def _handle_file_search_grounding_metadata_streaming(
+        self, grounding_metadata: GroundingMetadata | None
+    ) -> AsyncIterator[ModelResponseStreamEvent]:
+        """Handle file search grounding metadata for streaming responses.
+
+        Yields a BuiltinToolReturnPart if file search results are available in the grounding metadata.
+        """
+        if not self._file_search_tool_call_id or not grounding_metadata:
+            return
+
+        grounding_chunks = grounding_metadata.grounding_chunks
+        retrieved_contexts = _extract_file_search_retrieved_contexts(grounding_chunks)
+        if retrieved_contexts:  # pragma: no branch
+            yield self._parts_manager.handle_part(
+                vendor_part_id=uuid4(),
+                part=BuiltinToolReturnPart(
+                    provider_name=self.provider_name,
+                    tool_name=FileSearchTool.kind,
+                    tool_call_id=self._file_search_tool_call_id,
+                    content=retrieved_contexts,
+                ),
+            )
+            self._file_search_tool_call_id = None
+
+    def _handle_executable_code_streaming(
+        self, executable_code: ExecutableCode
+    ) -> tuple[ModelResponsePart | None, str | None]:
+        """Handle executable code for streaming responses.
+
+        Returns a tuple of (part_obj, code_execution_tool_call_id).
+        part_obj is None if the part was already generated (to avoid duplicates).
+        code_execution_tool_call_id is None for file_search, or the new ID for code_execution.
+        """
+        code = executable_code.code
+        if not code:  # pragma: no cover
+            code_execution_tool_call_id = _utils.generate_tool_call_id()
```

Comment:
These lines are identical to the `else` below, so I only want to have it once. So then I think the next `if has_file_search_tool...` can be `if code and ...`